src/System.Management.Automation/engine/parser/PSType.cs

```diff
@@ -1091,10 +1092,29 @@ internal static List<DefineEnumHelper> Sort(List<DefineEnumHelper> defineEnumHel
 
             internal void DefineEnum()
             {
+                var underlyingType = typeof(int);
+                foreach (var type in _enumDefinitionAst.BaseTypes)
+                {
+                    var resolvedType = type.TypeName.GetReflectionType();
+                    if (resolvedType == null || !_validBaseTypes.Contains(resolvedType))
```

Comment:
I think we could do the check in parser. Seems it is more useful.

Reply:
Our goal is to move as many checks to the parser as possible. These checks are no longer needed in PSTypes.